pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -65,7 +66,7 @@ func checkBundleCached() error {
 	return nil
```

Comment:
This is only checking for `~/.crc/cache/crc_libvirt_4.6.1.crcbundle`, this needs to start with a check for `~/.crc/cache/crc_libvirt_4.6.1/`, or something more sophisticated which checks the directory has a valid bundle.

Reply:
yes, more generally the name of the bundle is both coming from the crcbundle file on disk and from the directory in the bundle. This can cause mismatch sometimes for developers when the name is not the same. 
I guess we should revisit that.

At the end of the extraction we could also put a small empty file ".done" to indicate that the extraction is complete.

Reply:
Yeah my point was that the preflight check is not doing its job fully, if I `rm -rf ~/.crc/cache/crc_libvirt_4.6.1.crcbundle` and the run `crc setup`, the preflight check won't catch this, and `fixBundleExtracted` will not be called. There is a `bundle.Extract()` call in `getCrcBundleInfo()` which will get triggered and fix that, but it's not really meant for that situation.

Reply:
Imo this still misses a check which calls `GetCachedBundleInfo` and then makes sure the various files it lists exist.

Reply:
I will open a follow-up with this. It is a more massive refactoring.